### ISSUE TITLE
UI Customizations Don't Appear to Persist

### ISSUE THREAD

**Issue by gskerry** [keepassxreboot/keepassxc#10537]
## Overview
OS cache recently cleared. Had to reset customizations to UI. They don't appear to survive a restart or a database lock/unlock

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
1. (New install or clear OS cache) 
2. Open KP
3. Make UI customizations (e.g. sidebar width, visible columns & widths, etc)
4. Lock database
5. unlock database

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )
UI changes would persist and be the same as previous set

## Actual Behavior
[NOTE]: # ( Tell us what actually happens )
UI appears to 'reset' to a default view and layout.

## Context
[NOTE]: # ( Give us any additional information you may have. )
n/a

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - Version 2.7.7
Revision: 68e2dd8

[NOTE]: # ( Pick choices based on your environment )
Operating System: macOS

**Comment by phoerious** [comment]
I see that sometimes with the window size, but only on macOS. Need to investigate what's happening there.

**Comment by xboxones1** [comment]
Very rarely, but have encountered this problem on Windows and Linux. These are mainly column settings, visibility, width, etc.

**Comment by admiralspeedy** [comment]
I have given up hiding the columns I don't want to see because they reappear ever time I restart mt PC (Windows 11).

**Comment by droidmonkey** [comment]
Make sure you aren't clearing your %APPDATALOCAL% folder on every restart. You might also need to set your columns then manually EXIT keepassxc to ensure the settings are written.

**Comment by SdoggaMan** [comment]
This is still happening currently and quite frustrating. I've tried specifically closing the app first before rebooting (as elsewhere it was mentioned that the app may not save preferences when remaining open when the OS shuts down) and tried leaving it in focus, window open in background, and in taskbar. It seems to do it at random. I use KPXC on three Windows machines and there are no specific settings for %APPDATA%, just default Windows behaviour.

**Comment by droidmonkey** [comment]
This report is against macOS. If you keep seeing these problems, go to the application settings, scroll to the bottom, and click "Reset Settings". This will ensure a clean settings file. Qt seems to have an issue sometimes with the settings file and it cannot seem to recover from it without blasting it away.

**Comment by phoerious** [comment]
I don't think you need to wipe all your settings, but at least the local settings in `~/Library/Application Support/KeePassXC`. I also noticed that I had a `~/Library/Preferences/org.keepassxc.keepassxc.plist` still with geometry preferences. After deletion, it didn't come back.

**Comment by phoerious** [comment]
On a second glance, I think there's something wrong with the macOS settings on a more fundamental level. `~/Library/Application Support/KeePassXC` seems to contain all the settings, not just the local ones, whereas `~/Library/Preferences/KeePassXC` doesn't exist. Deleting the local settings, fixes the bug, but it also wipes out all your other settings.

**Comment by lkrms** [comment]
Is anyone else with this issue opening KeePassXC from the command line? I use the `--pw-stdin` option to unlock my password databases and in a quick test just now, was able to make window size and sidebar width persist when KeePassXC was opened directly; it was only when I went back to opening it via `KeePassXC.app/Contents/MacOS/KeePassXC` that geometry was lost. Mentioning in case it helps narrow down a cause.

**Comment by ozdreamern** [comment]
My variation of this issue occurs in Linux (Pop_OS 24.04 running vanilla GNOME 46), using KeePassXC via Flatpak.  In my scenario, KeePassXC remembers its window settings between sessions, UNTIL I make a change to the display geometry itself -- i.e., if I launch it after docking or un-docking my laptop to/from my external display, its window size resets to default.  Same behavior for years (under both Xorg and Wayland, I think, but I usually run Wayland these days).

**Comment by lkrms** [comment]
Thanks for your work on this issue. On macOS, at least (I haven't tested other builds yet), the same behaviour -- KeePassXC window geometry not being restored -- still occurs when KeePassXC is opened with `--pw-stdin`, whether minimizing at startup is enabled or not.

**Comment by phoerious** [comment]
The main issue was that Qt cannot apply geometry to hidden windows. This must be a different problem. I'll look into it.
